keylocker.go

```diff
@@ -0,0 +1,76 @@
+package atoms
+
+import "sync"
+
+// KeyLocker allows to hold locks by key.
+type KeyLocker struct {
+	m   map[string]*sync.RWMutex
+	mux sync.RWMutex
```

Comment:
When including a mux, it should be the very first declaration per our best practices